osgp/protocol-adapter-iec60870/osgp-protocol-adapter-iec60870/src/main/java/org/opensmartgridplatform/adapter/protocol/iec60870/infra/messaging/LogItemRequestMessageSender.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright 2019 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.adapter.protocol.iec60870.infra.messaging;
+
+import javax.jms.JMSException;
+import javax.jms.Message;
+import javax.jms.ObjectMessage;
+import javax.jms.Session;
+
+import org.opensmartgridplatform.shared.infra.jms.Constants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jms.core.JmsTemplate;
+import org.springframework.jms.core.MessageCreator;
+
+public class LogItemRequestMessageSender {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LogItemRequestMessageSender.class);
+
+    @Autowired
+    private JmsTemplate logItemRequestsJmsTemplate;
+
+    public void send(final LogItemRequestMessage logItemRequestMessage) {
+
+        LOGGER.debug("Sending LogItemRequestMessage");
+
+        this.logItemRequestsJmsTemplate.send(new MessageCreator() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this anonymous class number of lines from 21 to at most 20, or make it a named class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1188)